phoenix-core/src/main/java/org/apache/phoenix/calcite/rel/PhoenixClientProject.java

```diff
@@ -73,13 +70,10 @@ public QueryPlan implement(PhoenixRelImplementor implementor) {
         QueryPlan plan = implementor.visitInput(0, (PhoenixQueryRel) getInput());
         implementor.popContext();
         
-        
-        PhoenixSequence sequence = CalciteUtils.findSequence(this);
-        final SequenceManager seqManager = sequence == null ?
-                null : new SequenceManager(new PhoenixStatement(sequence.pc));
-        implementor.setSequenceManager(seqManager);
-        TupleProjector tupleProjector = project(implementor);
-        if (seqManager != null) {
+
```

Comment:
This entire wrapping logic should live in PhoenixToEnumerableConverter instead. Wrapping it with an iterator not a query plan would be better, something like:
{code}

        return new DelegateQueryPlan((QueryPlan) plan) {
            @Override
            public ResultIterator iterator() throws SQLException {
                ResultIterator iterator = iterator(DefaultParallelScanGrouper.getInstance());
                if (phoenixImplementor.getSequenceManager().getSequenceCount() > 0) {
                    iterator = new SequenceResultIterator(iterator, phoenixImplementor.getSequenceManager());
                }
                return iterator;
            }
            @Override
            public ExplainPlan getExplainPlan() throws SQLException {
{code}